### DISCUSSION TITLE
More exceptions in SDKs

### DISCUSSION THREAD

**Discussion by SirusCodes** [appwrite/appwrite#3188] [Ideas]
Hey, I have tried to use Appwrite with Flutter.
The only thing that bugged me was there is only single `AppwriteException`.
Tho it does the job but IMO it is not good for DX.

Adding a few more exceptions like:
- DocumentNotFoundException
- AuthenticationException
and so on

**Comment by maeddin** [comment]
It's true that this would be handy if you only want to catch certain exceptions (For this case I also miss more exceptions). For everything else, the information provided is actually sufficient. Which use case exactly is causing you problems?

**Reply by SirusCodes** [reply]
Stumbled across this issue when I was trying to do authentication and later found out that there is only one exception.

So thought it would be great to have more exceptions and opened this discussion.

**Reply by maeddin** [reply]
Yes, that is a problem. I have currently implemented my own exceptions for such purposes, which are thrown in the catch-block of the AppWrite exceptions, but this is of course not quite as nice.

**Reply by SirusCodes** [reply]
Exactly who likes an if-else ladder or switch cases :)
